src/caching/IPFSClient.ts

```diff
@@ -0,0 +1,82 @@
+import { buildIPFSClient, retrieveValueFromIPFS, storeValueInIPFS } from "../utils/IPFSUtils";
+import { CachingMechanismInterface } from "../interfaces";
+import { Struct } from "superstruct";
+import winston from "winston";
+import PinataClient from "@pinata/sdk";
+import { jsonReviverWithBigNumbers, jsonReplacerWithBigNumbers } from "../utils";
+
+/**
+ * A client for interacting with the IPFS. This is a wrapper around the IPFS API.
+ * This client also is a part of the caching mechanism interface.
+ * @see https://docs.ipfs.io/reference/http/api/
+ */
+export class IPFSClient implements CachingMechanismInterface {
```

Comment:
NIT: We are actually pretty opinionated on using the specific Pinata API here. So maybe renaming to `PinataIPFSClient` would be more accurate. The class `IPFSClient` then could be used in the future to implement a platform-agnostic IPFS HTTP Client that, for example, could connect directly to an IPFS node.

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/341/commits/79537dbf655bcac073d3400dca068adb8f25c3f3